### ISSUE TITLE
Build instructions do not mention cmake build or packaging

### ISSUE THREAD

**Issue by KantarBruceAdams** [protocolbuffers/protobuf#5760]
The build instructions are given as:

./configure
make
make install

Tl;Dr; I have several related issues around this:
1. autogen is not mentioned
2. cmake is not mentioned
3. creating packages is not mentioned
4. out of source builds are not discussed

Further details:

Direct installs from source can be a bad idea for several well known reasons
* interferes will vendor packaging (though source install go to /usr/local and package go to /usr
* they can awkward to uninstall (few projects provide a make uninstall - this one does not)

Are out of source builds officially supported (as would normally be recommended?)
i.e.
mkdir build
cd build
../configure
make
make install

There is no mention of the cmake way of compiling using:

mkdir build
cd build
cmake ../cmake
make
make install

Typically it is better to create a package yourself and then install that package.

I would prefer to run:

cpack -G RPM
sudo rpm -i *.rpm

On the autotools side there is no mention of running "make dist" to create a suitable RPM or .deb for your platform, assuming it does. I was not able to get this to work myself (I have been using fpm as a workaround for this as I have been unable to create packages for protocol buffers 3.3.0). It is unclear to me how the release distribution packages are created by the maintainers.

I think it would be  adding a note somewhere about what the relative capabilities and limitations are of the cmake build vs the autotools build, especially given #5124 

Also the configure script needs to be generated using ./autogen.sh if getting the source via git

**Comment by BSBandme** [comment]
I think we have some instruction for autogen [here](https://github.com/protocolbuffers/protobuf/blob/master/src/README.md)

And for cmake we have some instruction for [building with vs](https://github.com/protocolbuffers/protobuf/blob/master/cmake/README.md)

**Comment by KantarBruceAdams** [comment]
**But how are the packages created?** 

I will grant 1 and 2. There is a line about cmake and autogen on master (but not on the old 3.3.0 release I have been required to use).
The implication is that cmake support is only provided for windows. This is not true given  #5124.

Are out of source builds supported/allowed/encouraged or not recommended?
